tests/kernel/device/src/main.c

```diff
@@ -247,6 +268,110 @@ void test_dummy_device_pm(void)
 }
 #endif
 
+/* this is for storing sequence during initializtion */
+extern int init_level_sequence[4];
+extern int init_priority_sequence[4];
+extern unsigned int seq_level_cnt;
+extern unsigned int seq_priority_cnt;
+
+/**
+ * @brief Test initialization level for device driver instances
+ *
+ * @details After the defined device instances have initialized, we check the
+ * sequence number that each driver stored during initialization. If the
+ * sequence of initial level stored is corresponding with our expectation, it
+ * means assigning the level for driver instance works.
+ *
+ * @ingroup kernel_device_tests
+ */
+void test_device_init_level(void)
+{
+	bool seq_correct = true;
+
+	/* we check if the stored pexecuting sequence for different level is
```

Comment:
`pexecuting`: typo?

Reply:
Sorry for reply late, miss it. Yes, it's a typo and got it fixed, thanks!